### PR TITLE
Add optional error to readAsTextAsync callback in adm-zip

### PR DESCRIPTION
**Pull request by JakeGinnivan**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/cthackers/adm-zip/blob/master/zipEntry.js#L63
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/adm-zip/index.d.ts [hunk 1]
```diff
@@ -65,14 +65,14 @@ declare class AdmZip {
      * @param callback Called with the resulting string.
      * @param encoding Optional. If no encoding is specified utf8 is used
      */
-    readAsTextAsync(fileName: string, callback: (data: string) => any, encoding?: string): void;
+    readAsTextAsync(fileName: string, callback: (data: string, err?: any) => any, encoding?: string): void;
```
Comment (kevincharm) [inline types/adm-zip/index.d.ts]: type signature for second callback argument is `err?: string`, same below
ref: https://github.com/cthackers/adm-zip/blob/576f7c5a5ccd529c75c0c7cf870a69856d99ca3a/util/errors.js#L9

**Review by jvilk — CHANGES_REQUESTED** [review]
Is there ever a case where adm-zip calls the callback with an error? Looking at the source code, there isn't:

https://github.com/cthackers/adm-zip/blob/3f00a03ff55a1e1643b690a67fbe164e9ed7f48a/adm-zip.js#L113

Am I mistaken, or is this correct?

File: types/adm-zip/index.d.ts (hunk 1 above)
Comment (jvilk) [reply types/adm-zip/index.d.ts]: Actually, unless I'm missing something, it doesn't even look like adm-zip calls the callback with an error.

https://github.com/cthackers/adm-zip/blob/3f00a03ff55a1e1643b690a67fbe164e9ed7f48a/adm-zip.js#L113

**Comment by kevincharm** [comment]
@jvilk hi, I raised this issue to @JakeGinnivan. You are right, `readAsTextAsync` does not ever callback with a second error argument.

It does pass a callback down to `getDataAsync` but only checks for a non-empty `data` object: https://github.com/cthackers/adm-zip/blob/3f00a03ff55a1e1643b690a67fbe164e9ed7f48a/adm-zip.js#L116

Then `getDataAsync` passes down that callback to `decompress`: https://github.com/cthackers/adm-zip/blob/3f00a03ff55a1e1643b690a67fbe164e9ed7f48a/zipEntry.js#L245

Then `decompress` can invoke that callback with non-empty (!) `data` and an `error` string: https://github.com/cthackers/adm-zip/blob/3f00a03ff55a1e1643b690a67fbe164e9ed7f48a/zipEntry.js#L62

So it actually looks like an issue in the codebase itself, not in the type definitions. Sorry about that!

**Comment by jvilk** [comment]
@kevincharm Yeah, I think it makes sense to re-open if that change is merged. Until then, the change is technically invalid.

**Pull request closed without merging**